### ISSUE TITLE
renovation - DTH3_L1AD

### ISSUE THREAD

**Issue by ran-codes** [Drexel-UHC/salurbal-fair-renovations#61]

**Comment by usamabilal** [comment]
I see GHE3 codes as more strata than data attribute, since they refer to a "population" (people that died from cause X). What would be a justification for including them as data attributes?

**Comment by ran-codes** [comment]
another line of thought @usamabilal is we can have tiers as variables them selves? (e.g. make it searchable ... cause right now strata is not very searchable).

so instead of 1 variable (DTH3 with x number of strata) we have x number of variables DTH3GHE1, DTH3GHE2... eahc variable with own var_def... so people can find these more easily. thoughts @usamabilal

**Comment by usamabilal** [comment]
Not sure I follow, do you have an example of how this would look like? (a similar one in a different renovation)

**Comment by ran-codes** [comment]
this hasn't happened before. but something like [this example](https://github.com/Drexel-UHC/salurbal-fair-renovations/files/12304543/DTH-.single.vs.multiple.variables.xlsx). SO the question is should we organize cause specific DTH data as case 1 (single variable) or case 2 (multiple variables).

![image](https://github.com/Drexel-UHC/salurbal-fair-renovations/assets/45013044/8c08c199-d268-4ba8-af61-3f02f8bd26a4)


does this help?

**Comment by usamabilal** [comment]
@ran-codes and I just talked. We're going with case 1 (strata) since we dont mind people not being able to search for causes fo death when searching for variables, but they should be able to search for "cause of death" (or however we call the DTH1 varaible) or just mortality. Then cause of death will be a strata there

**Comment by juruchi** [comment]
Keeping GHE3 as strata results in about 5000 strata rows. I remember past conversations about censoring data with less than a certain number of counts, which could reduce the total number of rows visible on the data portal, but I'm not sure by how much.

**Comment by juruchi** [comment]
Talking with @goro-yamada, I've come to learn that level 3 data should all be suppressed due to low mortality counts for individual causes of death. Given this, do we need to go through the renovation process for this dataset?

**Comment by usamabilal** [comment]
I believe we’re renovating regardless of availability. Right ran ?
________________________________

**Comment by ran-codes** [comment]
Yes we should still renovate. The primary purpose of renovation/ETL is to make data/metadata machines actionable for internal use. So even if certain data is not going to public it is still import we have it machine actionable for internal use.

**Comment by juruchi** [comment]
@usamabilal @ran-codes I have come up with some code to programmatically fill in the strata_description column in the codebook for DTH1 (#59). That dataset has ~200 strata. The DTH2 dataset is expected to have ~900 strata and this one has ~5000. Before going ahead and populating the codebooks I'd like to confirm that the plan is to include GHE group codes as part of the dataset's strata.

**Comment by usamabilal** [comment]
Thanks @juruchi , just to confirm, what are the strata for DTH1 for example? 
3 causes (or 5 because it's not DTHRED?) x 2 sexes x 19 age groups x ...?

**Comment by juruchi** [comment]
5 GHE groups (3 causes + 1 code for missing ICD 10 + 1 code for ICD10 codes that are not in the GHE classification list) x 2 sexes x 19 age groups

**Comment by usamabilal** [comment]
Got it. Then yes, this is the correct number of strata for all. Good thing to do this programatically!

Separate issue: There should be no missing ICD-10 codes at all... R99 is the missing iCD-10 code (and it belongs to the ill-defined diseases section). Where are there missing ICD10 codes? Furthermore, every code should be in GHE. We chose it for that reason. We've reviewed codes not included there and assigned them to specific categories. Since this is DTH and not DTHRED, I was expecting the ill-defined diseases and injuries of ill-defined intent.

**Comment by juruchi** [comment]
SV has ~600 missing ICD10 records each year from 2001-2020. The records with an ICD10 code that is not in the list of GHE classification occur in BR, CL, CR, GT, MX for pretty much every year from 2000-2020. Each year there are 500-1000 observations with this code.

**Comment by usamabilal** [comment]
Thanks Jessica:

- On the "not included", we did a review of those with @goro-yamada and Jordan sometime ago and classified all into specific GHE categories. E.g., the COVID ones going into acute respiratory infections, etc. 
- All missing ICD-10 codes should be R99 and classified into the ill-defined category (and redistributed in the datasets where we redistribute).

The key reason of using an exhaustive classification such as GHE is that every death should be part of a category. 

Not sure if we need to pause renovation on DTH1/2/3 until we clarify this and make whatever corrections are needed. I am now seeing that there's a 99997 code for DTH1 in SV, and 99999 for CL. We should only have codes that are included in GHE, and GHE already accomodates missing data (R99 -> 1610) and we recoded all existing codes taht were not part of GHE to something.

**Comment by juruchi** [comment]
After reviewing the data again, and talking with Goro and Jordan, we found that all instances of strata including the ```99999``` code have a value of 0 (no deaths attributed to this code). We also found that the presence of ```99999``` is inconsistent across country as it appears in BR, CL, CR, GT, and MX but not in the other countries. Given this, we propose dropping strata with the ```99999``` code from the renovation files.

Regarding the ```99997``` code, we found an inconsistency in the number of deaths attributed to this code when reading directly from the data in the unencrypted folder and using the data put together as part of the renovation process. For instance, reading from ```DTH3SV_L1AD_20230515.csv``` gives 608 ```99997``` deaths, while reading from the ```i$raw_data``` object in ```etl_DTH3_v1.0.R``` gives 129 ```99997``` deaths. @ran-codes are you aware of anything in the import step that could be causing this difference?

Jordan will provide additional comments about the ```99997``` code.

**Comment by ran-codes** [comment]
thanks for brining this up @juruchi. I'll take a look tomorrow morning and get back to you ASAP. Thanks!

**Comment by usamabilal** [comment]
Thanks @juruchi ! But i think we need to address this at the cleaning level: there shouldnt be any non-GHE codes in the files (regardless of whether part of renovation or not).

**Comment by ran-codes** [comment]
Thoughts to discuss in todays meeting. @usamabilal feel free to jumping in if oyu have time (i forwarded you the invite)

## `99999` for GH3/strata_3_value 

I agree with Usama, That in the original data it makes more sense to keep GH3 coding as the original GHE3 code when cases DTH3DEATH is equal to 0. We already know the estimate county is zero... but with the status quo data structure we don't know for what tier it is zero. am i understanding what you mean @usamabilal ? (see example below)

![image](https://github.com/Drexel-UHC/salurbal-fair-renovations/assets/45013044/ad56f1a3-325d-4567-97af-541ccd79d15f)

**These `99999` should just be the GHE3 tiers. so we can know which tier have a count of zero.**

## Data import subsets during development

@juruchi good catch about the inconsistency. This is because some datasets (mainly DTH3 is very big) and to keep things quick we only import a seeded subset when in development  (`env == 'dev'`). See line 135 of [get_setup_imports()](https://github.com/Drexel-UHC/salurbal-fair-renovations/blob/main/R/get_setup_imports.R). The whole data set is processed in production  (`env == 'prod'`)

that being said. for DTH increased the sample per file to 500,000 instead of the usual 50,000 rows just to be safe because of all the GHE3 tiers. Not sure if this will cause an issue during development. lmk what you think.

**Comment by goro-yamada** [comment]
Hi Usama,

It's me who made a decision. We understood what you have said and decided considering other factors, which include:

  *   We need to change file names for the aggregated datasets without redistribution, whose date will become inconsistent with file names of others. Also, we do not want to change the file name of others if there are no changes. This was the main reason.
  *   99999 means ICD is missing. This does not occur in the cause of death. I.e., There are no mortality records with GHE = 99999 (death counts under 99999 are always zero). So, there is no lie. However, you can argue that such concept (records without ICD) does not exist first of all.
  *   99999 is used for variables for underlying conditions as well. We see this code commonly in these variables since the number of underlying conditions differs by case. This is one reason of why 99999 appears in the aggregated datasets.
  *   But we got to know that 99999 was not used uniformly as Jess mentioned, which should be due to a coding error.

We talked within the team yesterday, and we decided as Jess mentioned. On the other hand, we did not have strong opinions about what to do. If you want us to remove records with 99999 in the original data, we can do so.

Goro

**Comment by goro-yamada** [comment]
I started wondering, how users can know if the data they download now are different from those downloaded some time ago. In the current system, we can know by looking at the filename. With the current data portal system, how can we?

**Comment by juruchi** [comment]
@jrjordanrh1992 Here is where we are discussing the DTH3 dataset

**Comment by jrjordanrh1992** [comment]
Hello all! 

About code 99997: This only happens in SV. We have 608 observations (mostly from 2019) which have ICD-10 codes that are not in our GHE list. As Usama mentioned, this has happened in causes of death like COVID or other specific cases, which all have been added to our GHE list. In the cases that remain in SV with 99997, these are ICD-10 codes that are not valid causes of death by themselves and start with "S" or "T". These kind of ICD-10 codes are usually found in the "Multiple causes of death" variables, but in this case they appear in the basic cause of death in SV, which we believe is an error in the raw data. Since they don't fit the categories of the GHE list, we left them with the code 99997 in the non-redistributed data. In the redistributed data we don't have 99997, since they get redistributed.

**Comment by usamabilal** [comment]
Thanks Goro, Jordan and Jessica.
Then, can we:

- No ICD Codes: if there are none, i'd just remove any mention to 99999. If there were any, I think these should be recoded to R99 (and classified accordingly in ICD10). R99 is the code for no code.
- "New icd codes" (COVID, etc): these have been added already to GHE, so nothing to worry about
- Invalid codes S/T: this is more "delicate", as this is clearly an error in their coding (other countries just use a second code for this like Chile, where you have to go to that second code to find the underlying cause of death: I imagine SV has not done this, right?). An alternative, instead of having to "break" the GHE classification (and have unusable ICD-10 codes in the ICD10 version) would be to just assign these to Y34 (the missing injury code) and classify accordingly.

**Comment by jrjordanrh1992** [comment]
Hi Usama,

About S/T: We checked the raw data of SV. There's only one variable with an ICD code for the cause of death, which we are using. However, there is a categorical variable which specifies 10 different categories of violent death (e.g. "ahogamiento", "arma blanca", etc.). All of the observations with “S/T” fall under one of these categories. Given this information:

1. We agree that is reasonable to assign Y34 to these cases and update our mortality data accordingly. 
2. We are going to share this with Alex so he can consider the information of the categorical variable for the new redistribution algorithm of injuries.

**Comment by usamabilal** [comment]
Sounds good! If we do the Y34 reassignment then there wouldnt be any 99999 and 99997 codes, right? (since everything will match GHE). thanks

**Comment by jrjordanrh1992** [comment]
That's correct Usama, we will not have 99999 and 99997. We will work on updating this next week.

**Comment by usamabilal** [comment]
perfecto, thanks!

**Comment by juruchi** [comment]
I am using ```//files.drexel.edu/encrypted/SOPH/UHC/Projects/SALURBAL/Data/Files/Mortality/_Processed_data/data_ICD/ghe_codes_tier3new.rdata``` to programmatically pull in names for DTHCOD_GHE3 codes and fill in the strata_description column in the codebook. There are some group values that are associated with more than one name. For these 3 cases, what should be the name used in strata_description? 

![image](https://github.com/Drexel-UHC/salurbal-fair-renovations/assets/75770025/9bff780d-bd7a-49ff-b504-56953603a478)

**Comment by juruchi** [comment]
@goro-yamada For each of the 3 group codes listed in the table above, which of the associated name values should be used to describe the group? This is the name that would ultimately be included in the codebook under strata_description.

**Comment by goro-yamada** [comment]
@juruchi Before I answer your question, are we trying to make the data available horizontally stratified by GHE or ICD? I do not think users will need data in such a format. Are we preparing a codebook for individual death records or aggregated death count data, by the way?

**Comment by juruchi** [comment]
@goro-yamada This dataset has been updated to include age-aggregated (stratified by sex) death counts with suppression for values 1-9. The codebook has been updated accordingly. Please use the links in the comment above to review the codebook and associated sources.

**Comment by goro-yamada** [comment]
@juruchi They looked okay to me.
